packages/google-pay-integration/src/google-pay-button-strategy.ts

```diff
@@ -0,0 +1,273 @@
+    private _currencyCode?: string;
+
+    constructor(
+        private _paymentIntegrationService: PaymentIntegrationService,
+        private _googlePayPaymentProcessor: GooglePayPaymentProcessor,
+    ) {}
+
+    async initialize(
+        options: CheckoutButtonInitializeOptions & WithGooglePayButtonInitializeOptions,
+    ): Promise<void> {
+        if (!options.methodId || !isGooglePayKey(options.methodId)) {
+            throw new InvalidArgumentError(
+                'Unable to proceed because "methodId" is not a valid key.',
+            );
+        }
+
+        this._methodId = options.methodId;
+
+        if (!options.containerId) {
+            throw new InvalidArgumentError(
+                'Unable to proceed because "containerId" is not a valid key.',
+            );
+        }
+
+        const googlePayOptions = options[this._getMethodId()];
+
+        if (!googlePayOptions) {
+            throw new InvalidArgumentError('Unable to proceed without valid options.');
+        }
+
+        const { buyNowInitializeOptions, currencyCode } = googlePayOptions;
+
+        try {
+            if (buyNowInitializeOptions) {
+                if (!currencyCode) {
+                    throw new InvalidArgumentError(
+                        `Unable to initialize payment because "options.currencyCode" argument is not provided.`,
+                    );
+                }
+
+                await this._googlePayPaymentProcessor.initialize(
+                    () =>
+                        this._paymentIntegrationService
+                            .getState()
+                            .getPaymentMethodOrThrow<GooglePayInitializationData>(
+                                this._getMethodId(),
+                            ),
+                    () => this._getGooglePayClientOptions(googlePayOptions, options.currencyCode),
+                    !!buyNowInitializeOptions,
+                );
+
+                this._buyNowInitializeOptions = buyNowInitializeOptions;
+                this._currencyCode = currencyCode;
+            } else {
+                await this._paymentIntegrationService.loadDefaultCheckout();
+                await this._googlePayPaymentProcessor.initialize(() =>
+                    this._paymentIntegrationService
+                        .getState()
+                        .getPaymentMethodOrThrow<GooglePayInitializationData>(this._getMethodId()),
+                );
+            }
+        } catch (error) {
+            return;
+        }
+
+        this._addPaymentButton({
+            ...googlePayOptions,
+            container: options.containerId,
+        });
+    }
+
+    deinitialize(): Promise<void> {
+        return Promise.resolve();
+    }
+
+    private _addPaymentButton({
+        container,
+        buttonColor,
+        buttonType,
+        onError,
+    }: GooglePayButtonInitializeOptions): void {
+        this._paymentButton =
+            this._paymentButton ??
```

Comment:
Do we have a place where this method called twice?

Reply:
Thanks. There is only one place, fixed it